hnn_core/batch_simulate.py

```diff
@@ -180,3 +282,31 @@ def _generate_param_combinations(self, param_grid, combinations=True):
             param_combinations = [dict(zip(keys, combination))
                                   for combination in zip(*values)]
         return param_combinations
+
+    def save(self, results, start_idx, end_idx):
```

Comment:
```suggestion
    def _save(self, results, start_idx, end_idx):
```
Perhaps this should be a private function since users should specify saving in the `__init__()`